File: packages-gen/js/libs/xt-ui/frames/detail.js
const ui = require("@xtalk/ui/core.js")

function view(frame,record){
  let fields = (frame["opts"])["fields"] || [];
  let children = [];
  for(let field of fields){
    let id = field["id"];
    children.push(ui.node("ui/row",{"class":"justify-between gap-4","key":id},[
      ui.node("ui/label",{"value":field["label"] || id},[]),
      ui.text(record[id] || "",{})
    ]));
  };
  return ui.node("ui/card-content",{"class":"flex flex-col gap-3"},children);
}

module.exports = {["view"]:view}

File: packages-gen/dart/libs/xt-ui/lib/frames/detail.dart
import 'package:xtalk_ui/core.dart' as ui;


view(frame, record) {
  var fields = (frame["opts"])["fields"] ?? <dynamic>[];
  var children = <dynamic>[];
  var arr_52834 = fields;
  for(var i52835 = 0; i52835 < arr_52834.length; ++i52835){
    var field = arr_52834[i52835];
    var id = field["id"];
    children.add(ui.node("ui/row",<dynamic, dynamic>{"class":"justify-between gap-4","key":id},<dynamic>[
      ui.node(
          "ui/label",
          <dynamic, dynamic>{"value":field["label"] ?? id},
          <dynamic>[]
        ),
      ui.text(record[id] ?? "",<dynamic, dynamic>{})
    ]));
  };
  return ui.node(
    "ui/card-content",
    <dynamic, dynamic>{"class":"flex flex-col gap-3"},
    children
  );
}